quinn-proto/src/connection/mod.rs

```diff
@@ -1908,6 +1917,23 @@ impl Connection {
                     trace!("dropping short packet during handshake");
                     return;
                 } else {
+                    if let (
+                        &Header::Initial { ref token, .. },
+                        &State::Handshake(state::Handshake {
+                            token: ref expected_token,
+                            ..
+                        }),
+                    ) = (&packet.header, &self.state)
```

Comment:
Oof, I often find multi-line `if let`s prety hard to follow. Is there some other way we can format this, maybe using some temporary named variables?

Reply:
Split into two `if let`s and dropped the `state::Handshake` destructuring. Much clearer, I agree.